Add time series histogram to the TSDB docs. (#4548)
elastic/docs-content: manage-data/data-store/data-streams/time-series-data-stream-tsds.md

Metrics are numeric measurements that change over time. Documents in a TSDS typically contain one or more metric fields. 

To mark a field as a metric, use the `time_series_metric` mapping parameter. This parameter ensures data is stored in an optimal way for time series analysis. The valid values for `time_series_metric` are `counter`, `gauge` and `histogram`:

`counter`

> :   A cumulative metric that only monotonically increases or resets to `0` (zero). For example, a count of errors or completed tasks that resets when a serving process restarts. A counter is supported by all [numeric field types](elasticsearch://reference/elasticsearch/mapping-reference/number.md)

Reviewer: Is this `int`, `long` and `double`?

Author: The mapping accepts all numeric fields, I have tried scaled_float, float, byte, short. I haven't tested it end to end though, I mean index data, downsample, run ESQL queries. Do you think this is something we need to explore further?

`gauge`
:   A metric that represents a single numeric that can arbitrarily increase or decrease. For example, a temperature or available disk space. A gauge is supported by all [numeric field types](elasticsearch://reference/elasticsearch/mapping-reference/number.md) and [`aggregate_metric_double`](elasticsearch://reference/elasticsearch/mapping-reference/aggregate-metric-double.md) (for internal use during downsampling, rarely user-populated).

> `histogram` {applies_to}`stack: preview 9.3`

Reviewer: Ditto.

> :   A metric that tracks the distribution of numerical values (like request times or sizes) by grouping them into "buckets" allowing you to monitor the shape of the data alongside the total count. A histogram is supported by [`histogram`](elasticsearch://reference/elasticsearch/mapping-reference/histogram.md) and [`exponential_histogram`](elasticsearch://reference/elasticsearch/mapping-reference/exponential-histogram.md). 

Reviewer: ```suggestion
:   A metric that tracks the distribution of numerical values, like latency or size distributions. A histogram is supported by [`histogram`](elasticsearch://reference/elasticsearch/mapping-reference/histogram.md) and [`exponential_histogram`](elasticsearch://reference/elasticsearch/mapping-reference/exponential-histogram.md). 
```
